assets/js/base/components/block-error-boundary/types.ts

```diff
@@ -47,7 +47,7 @@ export interface BlockErrorBoundaryProps extends BlockErrorBase {
 	 * Override the default error with a function that takes the error message and returns a React component
 	 */
 	renderError?: ( props: RenderErrorProps ) => React.ReactNode;
-	showErrorMessage?: boolean;
+	showErrorMessage?: boolean | unknown;
```

Comment:
> Should this be | undefined?

https://github.com/woocommerce/woocommerce-blocks/blob/c9f4a0ea2a4fa47a2374ff19767532507dc1683c/assets/js/base/components/block-error-boundary/types.ts#L50 affects https://github.com/woocommerce/woocommerce-blocks/blob/c69f13d7e6ab1b1ef7ba528e7b7015d6aacea179/assets/js/blocks/checkout/block.tsx#L187 whereas `CURRENT_USER_IS_ADMIN` is coming from https://github.com/woocommerce/woocommerce-blocks/blob/c69f13d7e6ab1b1ef7ba528e7b7015d6aacea179/assets/js/settings/shared/default-constants.ts#L12

As `allSettings.currentUserIsAdmin` has the type `unknown`, I was using 

```tsx
showErrorMessage?: boolean | unknown;
```

instead of

```tsx
showErrorMessage?: boolean | undefined;
```

Would you say it's better to change 

```tsx
export const CURRENT_USER_IS_ADMIN = allSettings.currentUserIsAdmin;
```

to

```tsx
export const CURRENT_USER_IS_ADMIN = allSettings.currentUserIsAdmin as boolean;
```

in https://github.com/woocommerce/woocommerce-blocks/blob/c69f13d7e6ab1b1ef7ba528e7b7015d6aacea179/assets/js/settings/shared/default-constants.ts, @mikejolley ?

Reply:
Yes, I think so. We know that it will be a boolean if provided. It technically could also be undefined I guess but under normal circumstances its a boolean.